File: LENTABP10/src/main/java/com/lenta/lentabp10/MobileObjectsService/IMboService.java
package com.lenta.lentabp10.MobileObjectsService;

import com.lenta.lentabp10.MobileObjectsService.models.MB_S_07;
import com.lenta.lentabp10.MobileObjectsService.models.MB_S_14;
import com.lenta.lentabp10.MobileObjectsService.models.MB_S_22;
import com.lenta.lentabp10.MobileObjectsService.models.MB_S_25;
import com.lenta.lentabp10.MobileObjectsService.models.MB_S_26;
import com.lenta.lentabp10.MobileObjectsService.models.MB_S_30;
import com.lenta.lentabp10.MobileObjectsService.models.MB_S_31;
import com.lenta.lentabp10.MobileObjectsService.models.MB_S_32;
import com.lenta.lentabp10.MobileObjectsService.models.MB_S_33;
import com.lenta.lentabp10.MobileObjectsService.models.MB_S_34;
import com.lenta.lentabp10.MobileObjectsService.models.MB_S_36;

import java.util.List;

public interface IMboService {

    //region MB_S_07
    MB_S_07 getUomInfo(String uom);
    //endregion

    //region MB_S_14
    List<MB_S_14> getParametersByParamName(String paramName);
    String getFirstParameterValueByName(String paramName);
    //endregion

    //region MB_S_22
    List<MB_S_22> getSetItemsByMatnrOsn(String matnrOsn);
    //endregion

    //region MB_S_25
    MB_S_25 getBarcodeInfo(String barcode);
    //endregion

    //region MB_S_26
    List<MB_S_26> getAllPrinters();
    //endregion

    //region MB_S_30
    //MB_S_30 getByMaterialNumber(string material);
    MB_S_30 getProductByMaterialNumber(String material);
    //endregion

    //region MB_S_31
    MB_S_31 getWriteOffCause(String taskType, String sectionId, String matkl, String ekgrp);
    //endregion

    //region MB_S_32
    List<MB_S_32> getWriteOffCauseByTask(String taskType);
    //endregion

    //region MB_S_33
    List<MB_S_33> getStoragesByTaskTypeAndTK(String taskType, String tkNumber);
    //endregion

    //region MB_S_34
    List<MB_S_34> getProductTypesByTaskType(String taskType);
    //endregion

    //region MB_S_36
    MB_S_36 getGisControlsByTaskCntrl(String taskCntrl);
    //endregion
}

File: writeoff_breakage/src/main/java/com/lenta/bp10/mobileObjectsService/IMboService.kt
package com.lenta.bp10.mobileObjectsService

import com.lenta.bp10.mobileObjectsService.models.*

interface IMboService {

    //region MB_S_07
    fun getUomInfo(uom: String): MB_S_07
    //endregion

    //region MB_S_14
    fun getParametersByParamName(paramName: String): List<MB_S_14>
    fun getFirstParameterValueByName(paramName: String): String
    //endregion

    //region MB_S_22
    fun getSetItemsByMatnrOsn(matnrOsn: String): List<MB_S_22>
    //endregion

    //region MB_S_25
    fun getBarcodeInfo(barcode: String): MB_S_25
    //endregion

    //region MB_S_26
    fun getAllPrinters(): List<MB_S_26>
    //endregion

    //region MB_S_30
    //MB_S_30 getByMaterialNumber(string material);
    fun getProductByMaterialNumber(material: String): MB_S_30
    //endregion

    //region MB_S_31
    fun getWriteOffCause(taskType: String, sectionId: String, matkl: String, ekgrp: String): MB_S_31?
    //endregion

    //region MB_S_32
    fun getWriteOffCauseByTask(taskType: String): List<MB_S_32>
    //endregion

    //region MB_S_33
    fun getStoragesByTaskTypeAndTK(taskType: String, tkNumber: String): List<MB_S_33>
    //endregion

    //region MB_S_34
    fun getProductTypesByTaskType(taskType: String): List<MB_S_34>
    //endregion

    //region MB_S_36
    fun getGisControlsByTaskCntrl(taskCntrl: String): MB_S_36?
    //endregion
}